File: beads.pas
{
  AUTHOR:wind1740
  LANG:PASCAL
  PROG:beads
}

var last,now,i,j,num:longint;
    flag:boolean;
	max:longint;
	chain:ansistring;
	w:longint;
    r:char;

procedure check;
begin
  if ((flag=true) and (r='b')) or ((flag=false) and (r='r')) then 
  begin
    //writeln (last,' ',now);
	if max<(last+now) then max:=last+now;
	last:=now-w;
	now:=w+1;
	w:=0;
	flag:=not flag;
  end
  else
  if r='w' then
  begin
	inc(w);
	inc(now);
  end
  else
  begin
	w:=0;
	inc(now);
  end;
end;	
	
begin
  assign (input,'beads.in');
  assign (output,'beads.out');
  reset (input);
  rewrite (output);
  
  flag:=true;
  max:=0;
  readln (num);
  readln (chain);
 
  for j:=1 to 2 do
    for i:=1 to num do
    begin
      r:=chain[i];   
      check;
    end;
  if max<(last+now) then max:=(last+now) div 2;
  if max>i then max:=i; 
  writeln (max);
  close (input);
  close (output);
end.
  

File: beads.java
/*
  AUTHOR:wind1740
  LANG:JAVA
  PROG:beads
*/
import java.io.*;
import java.util.*;

public class beads
{
  private static int last,now,num;
  private static boolean flag;
  private static int	maxx;
  private static String chain;
  private static int	w;
  private static char r;

  public static void check()
  {
    if (((flag==true)&&(r=='b'))||((flag==false)&&(r=='r')))
    {
      //writeln (last,' ',now);
	  if (maxx<(last+now))
         maxx=last+now;
	  last=now-w;
	  now=w+1;
	  w=0;
	  flag=!flag;
    }
    else
    if (r=='w')
    {
	  w++;
	  now++;
    }
    else
    {
	  w=0;
	  now++;
    }
  }	
	
  public static void main(String[] args) throws FileNotFoundException
  {
    Scanner sc=new Scanner(new File ("beads.in"));
    PrintWriter pw=new PrintWriter(new File("beads.out"));
  
    flag=true;
    maxx=0;
    num=sc.nextInt();
    sc.nextLine();
    chain=sc.nextLine();
 
    for (int j=1;j<=2;j++)
      for (int i=1;i<=num;i++)
      {
        r=chain.charAt(i-1);   
        check();
      }
    if (maxx<(last+now)) 
  	  maxx=(int)(last+now)/2;
    if (maxx>num)
	  maxx=num; 
    pw.println (maxx);
    sc.close();
    pw.close();
  }
}
  